### ISSUE TITLE
JDK zlib decompression support

### ISSUE THREAD

**Issue by phansson** [netty/netty#1481]
With the changes implemented in JDK 7 I'm a bit uncertain as to whether a dependency on JZLib is still required.

Currently Netty implements `JdkZlibEncoder` but doesn't implement a corresponding `JdkZlibDecoder`. I believe this is a historical omission based on the inability of previous versions of the JDK.

Source code for [ZlibCodecFactory](http://netty.io/4.0/xref/io/netty/handler/codec/compression/ZlibCodecFactory.html) gives some idea as to what is currently going on but no comments as to why there's no `JdkZlibDecoder`.

In other words: The proposal here is to implement `JdkZlibDecoder` in Netty and make it the default for decoding when using Java 7.
